external/vulkancts/modules/vulkan/conditional_rendering/vktConditionalDispatchTests.cpp

```diff
@@ -0,0 +1,359 @@
+	void						recordDispatch					(const vk::DeviceInterface&	vk,
+																 vk::VkCommandBuffer cmdBuffer,
+																 compute::Buffer& indirectBuffer);
+
+protected:
+	const DispatchCommandType		m_command;
+	const int						m_numCalls;
+	const ConditionalData			m_conditionalData;
+};
+
+ConditionalDispatchTest::ConditionalDispatchTest(tcu::TestContext&	testCtx,
+												 const std::string&	name,
+												 const std::string&	description,
+												 const ConditionalTestSpec& testSpec)
+	: TestCase		(testCtx, name, description)
+	, m_testSpec	(testSpec)
+{
+}
+
+void ConditionalDispatchTest::initPrograms (vk::SourceCollections& sourceCollections) const
+{
+	std::ostringstream src;
+	src << "#version 310 es\n"
+		<< "layout(local_size_x = 1u, local_size_y = 1u, local_size_z = 1u) in;\n"
+		<< "layout(set = 0, binding = 0, std140) buffer Out\n"
+		<< "{\n"
+		<< "    coherent uint count;\n"
+		<< "};\n"
+		<< "void main(void)\n"
+		<< "{\n"
+		<< "	atomicAdd(count, 1u);\n"
+		<< "}\n";
+
+	sourceCollections.glslSources.add("comp") << glu::ComputeSource(src.str());
+}
+
+TestInstance* ConditionalDispatchTest::createInstance (Context& context) const
+{
+	return new ConditionalDispatchTestInstance(context, m_testSpec);
+}
+
+ConditionalDispatchTestInstance::ConditionalDispatchTestInstance (Context &context, ConditionalTestSpec testSpec)
+	: TestInstance(context)
+	, m_command(testSpec.command)
+	, m_numCalls(testSpec.numCalls)
+	, m_conditionalData(testSpec.conditionalData)
+{
+	checkConditionalRenderingCapabilities(context, m_conditionalData);
+};
+
+void ConditionalDispatchTestInstance::recordDispatch (const vk::DeviceInterface& vk,
+													  vk::VkCommandBuffer cmdBuffer,
+													  compute::Buffer& indirectBuffer)
+{
+	for (int i = 0; i < m_numCalls; i++)
+	{
+		switch (m_command)
+		{
+			case DISPATCH_COMMAND_TYPE_DISPATCH:
+			{
+				vk.cmdDispatch(cmdBuffer, 1, 1, 1);
+				break;
+			}
+			case DISPATCH_COMMAND_TYPE_DISPATCH_INDIRECT:
+			{
+				vk.cmdDispatchIndirect(cmdBuffer, *indirectBuffer, 0);
+				break;
+			}
+			case DISPATCH_COMMAND_TYPE_DISPATCH_BASE:
+			{
+				vk.cmdDispatchBase(cmdBuffer, 0, 0, 0, 1, 1, 1);
+				break;
+			}
+			default: DE_ASSERT(DE_FALSE);
+		}
+	}
+}
+
+tcu::TestStatus ConditionalDispatchTestInstance::iterate (void)
+{
+	const vk::DeviceInterface&	vk					= m_context.getDeviceInterface();
+	const vk::VkDevice			device				= m_context.getDevice();
+	const vk::VkQueue			queue				= m_context.getUniversalQueue();
+	const deUint32			    queueFamilyIndex	= m_context.getUniversalQueueFamilyIndex();
+	vk::Allocator&				allocator			= m_context.getDefaultAllocator();
+
+	// Create a buffer and host-visible memory for it
+
+	const vk::VkDeviceSize bufferSizeBytes = sizeof(deUint32);
+	const compute::Buffer outputBuffer(vk, device, allocator, compute::makeBufferCreateInfo(bufferSizeBytes, vk::VK_BUFFER_USAGE_STORAGE_BUFFER_BIT), vk::MemoryRequirement::HostVisible);
```

Comment:
Aren't you assuming the outputBuffer is initialized to 0? Please init it to 0.

Reply:
Yes, done.